src/Internal/ResolvedEventParser.php

```diff
@@ -87,6 +87,6 @@ public static function parse(array $entry): ResolvedEvent
             DateTime::create($entry['updated'])
         );
 
-        return new ResolvedEvent($record, $link, null);
+        return new ResolvedEvent($linkedEvent, $record, null);
```

Comment:
Hmm actually this will maybe break the simple case without link. So I'm guessing we need this?

```
return new ResolvedEvent($linkedEvent ?? $record, $linkedEvent ? $record : null, null);
```